fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/SelectRowSource.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * (C) Copyright IBM Corp. 2019,2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query;
+
+import com.ibm.fhir.database.utils.query.expression.StatementRenderer;
+import com.ibm.fhir.database.utils.query.expression.StringStatementRenderer;
+
+/**
+ * Represents a table referenced in the from list
+ */
+public class SelectRowSource implements RowSource {
+
+    // The sub-query
+    private final Select select;
+
+    /**
+     * Protected constructor
+     *
+     * @param subQuery
+     * @param alias
+     */
+    protected SelectRowSource(Select subQuery) {
+        this.select = subQuery;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder();
+        result.append("(");
+        result.append(select.toString());
+        result.append(")");
+
+        return result.toString();
+    }
+
+    @Override
+    public String toPrettyString(boolean pretty) {
+
+        // TODO get rid, because we don't have a translator so can't really render
+        // a correct string
```

Comment:
does this apply to the prior? 

is there an example of the rendering?

Reply:
The toString() method is there to be useful for debugging, not an actual rendered version of the string. The toPrettyString was an attempt to make that debug statement a little more readable by using the renderer (which can handle some basic pretty-printing). There's some future work here as we get more comfortable with using this stuff. Leaving this here for now.

Reply:
When we refactor Select in #2321 this will get cleaned up